pgjdbc/src/main/java/org/postgresql/util/LruCache.java

```diff
@@ -116,11 +165,8 @@ public LruCache(int maxSizeEntries, long maxSizeBytes, boolean accessOrder,
    * @throws SQLException if entry creation fails
    */
   public synchronized Value borrow(Key key) throws SQLException {
-    Value value = cache.remove(key);
+    @Nullable Value value = cache.remove(key);
     if (value == null) {
-      if (createAction == null) {
-        throw new UnsupportedOperationException("createAction == null, so can't create object");
-      }
       return createAction.create(key);
```

Comment:
What is the reason for the change?

Reply:
It has nothing to do with the defect being fixed, and I can revert for create and/or evict if desired.

With lambdas it is simple to (concisely) provide default implementations.
It also seemed to make the evict and borrow methods easier to read. But that is certainly subjective. Eliminating the null check avoids a few bytecodes from the methods, which /might/ help with inlining, but the main advantage is clarity.
Let me know if you want reverted.

Reply:
Personally, I break all my "Java new feature" commits to their own. Often it doesn't matter, but a few times it's allowed me to easily revert them when I realized we need to abort. The new Java is a problem in some way, and we still need to support the old Java. :-)